### ISSUE TITLE
Invalid protocol version (1)

### ISSUE THREAD

**Issue by s4kibs4mi** [emqx/emqx#1972]
#### Environment

- OS: Ubuntu Bionic (docker/k8s)
- Erlang/OTP: 21
- EMQ: 3.0-rc.3

#### Description

When running in cluster mode sometimes it's throwing this error.
##### Broker log,
```
(emqx@10.12.2.12)1>
19:58:58.259 [debug] MQTT(10.128.0.2:53878): RECV <<16, 248, 2, 0, 6, 77, 81, 73, 115, 100, 112, 3, 194, 0, 60, 0, 20, 79, 116, 104, 101, 114, 95, 66, 68, 89, 74, 67, 75, 71, 86, 81, 69, 72, 85, 65, 80, 0, 6, 49, 50, 51, 52, 53, 54, 1, 76, 101, 121, 74, ...>>
(emqx@10.12.2.12)1>
19:58:58.259 [debug] [MQTT] RECV CONNECT(Q0, R0, D0, ClientId=Other_BDYJCKGVQEHUAP, ProtoName=MQIsdp, ProtoVsn=3, CleanStart=true, KeepAlive=60, Username=123456, Password=******)
(emqx@10.12.2.12)1> [
  "check client",
  " : ",
  %{
    client_id: "Other_BDYJCKGVQEHUAP",
    peername: {{10, 128, 0, 2}, 53878},
    username: "123456",
    zone: :external
  }
]
(emqx@10.12.2.12)1>
19:58:59.233 [error] [MQTT] Failed to open session: :client_id_unavailable
(emqx@10.12.2.12)1> [
  "on_client_connected client",
  " : ",
  %{
    client_id: "Other_BDYJCKGVQEHUAP",
    peername: {{10, 128, 0, 2}, 53878},
    username: "123456",
    zone: :external
  }
]
(emqx@10.12.2.12)1> ["on_client_connected connect_code", " : ", 128]
(emqx@10.12.2.12)1> [
  "on_client_connected connect_info",
  " : ",
  [
    zone: :external,
    client_id: "Other_BDYJCKGVQEHUAP",
    username: "123456",
    peername: {{10, 128, 0, 2}, 53878},
    peercert: :nossl,
    proto_ver: 3,
    proto_name: "MQIsdp",
    clean_start: true,
    keepalive: 60,
    mountpoint: :undefined,
    is_super: false,
    is_bridge: false,
    connected_at: {1542, 657538, 259703}
  ]
]
(emqx@10.12.2.12)1> ["on_client_connected env", " : ", []]
(emqx@10.12.2.12)1>
19:58:59.234 [debug] [MQTT] SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=1)
(emqx@10.12.2.12)1>
19:58:59.234 [debug] MQTT(10.128.0.2:53878): SEND <<32, 2, 0, 1>>
(emqx@10.12.2.12)1>
19:58:59.234 [debug] MQTT(10.128.0.2:53878): Terminated for {:shutdown, :unacceptable_protocol_version}
(emqx@10.12.2.12)1> ["Subscription failed - error", " : ", :error]
(emqx@10.12.2.12)1> ["Subscription failed - reason", " : ", :session_not_found]
```

##### Client log,
```
Exception in thread "main" Invalid protocol version (1)
	at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:31)
	at org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:988)
	at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:145)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)

```

Tried by setting version `MQTT_VERSION_3_1` and `MQTT_VERSION_3_1_1` both have similiar issue.

**Comment by tigercl** [comment]
@s4kibs4mi Your protocol version is fine, this issue is caused by the following reason:
> [error] [MQTT] Failed to open session: :client_id_unavailable

We converted the reason code for MQTT 3.1 so that the reason code becomes 'unacceptable_protocol_version'.

So back to the error itself, it will appear when clients with the same client id send CONNECT packet to server at the same time. Do you meet the above situation?

**Comment by s4kibs4mi** [comment]
@tigercl no. I was trying to connect reconnect only one connection and this is happening in cluster mode only. Is there any situation that broker taking time to clean session across cluster and in the mean time I am trying to reconnect. As I am using same client id every time, broker may consider them as same client id trying to connect as previous is not cleaned yet.

**Comment by tigercl** [comment]
@s4kibs4mi Can you provide more information such as what strategy of node discovery and autocluster you used?

**Comment by s4kibs4mi** [comment]
I am using kubernetes on GKE with these settings,
```
env:
          - name: EMQ_NAME
            value: "emqx"
          - name: EMQ_CLUSTER__DISCOVERY
            value: "k8s"
          - name: EMQ_CLUSTER__K8S__APISERVER
            value: "https://kubernetes.default:443"
          - name: EMQ_CLUSTER__K8S__NAMESPACE
            value: "emqx"
          - name: EMQ_CLUSTER__K8S__SERVICE_NAME
            value: "emqx"
          - name: EMQ_CLUSTER__K8S__ADDRESS_TYPE
            value: "ip"
          - name: EMQ_CLUSTER__K8S__APP_NAME
            value: "emqx"
          - name: EMQ_NODE__COOKIE
            value: "secret123"
          - name: EMQ_CLUSTER__NAME
            value: "emqx"
          - name: EMQ_CLUSTER__AUTOHEAL
            value: "on"
          - name: EMQ_CLUSTER__AUTOCLEAN
            value: "5m"
```

**Comment by tigercl** [comment]
Sorry for this problem in the previous version, and the bug has been fixed in [emqx v3.0-rc.4](https://github.com/emqx/emqx/releases)
